File: retos-python/17_busca_bombas.py
# Resultado 2 ⭐
def detect_bombs(grid):
    def count_bombs(i, j):
        bombs = 0
        for r in range(-1, 2):
            for c in range(-1, 2):
                if r == 0 and c == 0:
                    continue
                x, y = i + r, j + c
                if 0 <= x < len(grid) and 0 <= y < len(grid[0]) and grid[x][y]:
                    bombs += 1
        return bombs

    return [[count_bombs(i, j) for j in range(len(row))] for i, row in enumerate(grid)]

File: retos-ts/17_busca_bombas.ts
// Resultado 5 ⭐ 
function detectBombs(grid: boolean[][]): number[][]{    
    return grid.map((row,i) => {
        return row.map((_,j) => {
            let bombs: number = 0

            for (let r = -1; r < 2; r++) {
                for (let c = -1; c < 2; c++) {
                    if(c === 0 && r === 0) continue;
                    const x: number = i + r
                    const y: number = j + c
             
                    if(grid[x] && grid[x][y]) bombs++
       
                }
            }

            return bombs
        })
    })

}